### PR TITLE
Fix mungedocs for absolute paths

### LINKED ISSUE #11111: ./hack/run-gendocs.sh busted at HEAD, breaks pre-commits
#11038 broke `hack/gendocs.sh` for me. I can't generate non-relative links now, so precommits on HEAD are failing for me:

```
zml@zml:~/kubernetes$ ./hack/run-gendocs.sh
+++ [0711 06:35:01] Building go targets for linux/amd64:
    cmd/gendocs
    cmd/genman
    cmd/genbashcomp
    cmd/mungedocs
+++ [0711 06:35:06] Placing binaries
/usr/local/google/home/zml/kubernetes/docs/design/access.md
----
check-links:
contents were modified

[...]
```

Diff:
```
diff --git a/docs/design/access.md b/docs/design/access.md
index 85b9c8e..f0b77f6 100644
--- a/docs/design/access.md
+++ b/docs/design/access.md
@@ -151,7 +151,7 @@ In the Simple Profile:
 
 Namespaces versus userAccount vs Labels:
 - `userAccount`s are intended for audit logging (both name and UID should be logged), and to define who has access to `namespace`s.
-- `labels` (see [docs/labels.md](../../docs/labels.md)) should be used to distinguish pods, users, and other objects that cooperate towards a common goal but are different in some way, such as version, or responsibilities.
+- `labels` (see [/usr/local/google/home/zml/kubernetes/docs/labels.md](../../docs/labels.md)) should be used to distinguish pods, users, and other objects that cooperate towards a common goal but are different in some way, such as version, or responsibilities.
 - `namespace`s prevent name collisions between uncoordinated groups of people, and provide a place to attach common policies for co-operating groups of people.
 
 ```

I'm not sure why my environment is a special snowflake. Trying in a clean repo now.

### PR DESCRIPTION
**Pull request by zmerlynn**
chdir into the parent of the docs directory, then walk docs.

Fixes #11111

### CODE REVIEW

**Comment by zmerlynn** [comment]
cc @lavalamp @thockin @erictune

**Comment by zmerlynn** [comment]
The Shippable failure is #11105.

**Comment by zmerlynn** [comment]
I think I like my approach better than #11085's, for being way simpler. I considered the two argument approach, but just doing a single argument and a split requires a ton less explanation, and the `chdir` is less fraught with peril than string based path trimming, but @lavalamp is clearly trying to do more in his PR as well.

File: cmd/mungedocs/mungedocs.go [hunk 1]
```diff
@@ -127,6 +128,19 @@ func main() {
 		os.Exit(1)
 	}
 
+	// Split the root dir of "foo/docs" into "foo" and "docs". We
+	// chdir into "foo" and walk "docs" so the walk is always at a
+	// relative path.
+	root, docs := path.Split(*rootDir)
```
Comment (thockin) [inline cmd/mungedocs/mungedocs.go]: why not 

```
stem, leaf := path.Split(strings.TrimRight(*rootDir, "/"))
```

**Comment by thockin** [comment]
LGTM, I patched this into my client and confirm it works

**Comment by zmerlynn** [comment]
Thanks! Nit fixed, PTAL.

**Comment by thockin** [comment]
commiting to fix head.

**Pull request merged**